operator/controllers/lifecycle/keptntaskdefinition/controller.go

```diff
@@ -58,13 +58,39 @@ func (r *KeptnTaskDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.
 		r.Log.Error(err, "Failed to get the KeptnTaskDefinition")
 		return ctrl.Result{Requeue: true, RequeueAfter: 30 * time.Second}, nil
 	}
+	defSpec := common.GetRuntimeSpec(definition)
+	if definition.Spec.Container == nil && defSpec != nil { //if the spec is well-defined
```

Comment:
As we have validation webhook in place, it's enough to check if the defSpec is not nil, no need to check the container, as that cannot happen
```suggestion
	if defSpec != nil { //if the spec is well-defined
```

Reply:
but the validation is not merged yet and tests would fail without this 😢